tests/Stache/BasicStoreTest.php

```diff
@@ -35,16 +35,17 @@ public function it_gets_an_item_by_key()
     }
 
     /** @test */
-    public function items_are_different_instances_every_time()
+    public function items_are_the_same_instance_every_time()
     {
-        config(['cache.default' => 'file']); // Doesn't work when they're arrays since the object is stored in memory.
+        config(['cache.default' => 'file']);
         \Illuminate\Support\Facades\Cache::clear();
 
         file_put_contents($this->tempDir.'/foo.yaml', '');
 
         $this->assertNotNull($one = $this->store->getItem('foo'));
         $this->assertNotNull($two = $this->store->getItem('foo'));
-        $this->assertNotSame($one, $two);
+
+        $this->assertSame($one, $two);
```

Comment:
This test was intentional. We want different instances every time. Please revert this change.

See https://github.com/statamic/cms/commit/20d2c4cc50d316368e888d0882e17f2b6c7dc6c2

Reply:
Hi, now getItem returns the item from blink (which is in-memory). We would expect the result to be exactly the same. I think that's right?

Reply:
Try

```php
    /** @test */
    public function items_are_different_instances_every_time()
    {
        config(['cache.default' => 'file']); // Doesn't work when they're arrays since the object is stored in memory.
        \Illuminate\Support\Facades\Cache::clear();

        file_put_contents($this->tempDir.'/foo.yaml', '');

        $this->assertNotNull($one = $this->store->getItem('foo'));
        $this->store->forgetItem('foo');

        $this->assertNotNull($two = $this->store->getItem('foo'));
        $this->assertNotSame($one, $two);
    }
```

Reply:
Could have the both tests? if its worth it?

```php
    /** @test */
    public function items_are_different_instances_every_time()
    {
        config(['cache.default' => 'file']); // Doesn't work when they're arrays since the object is stored in memory.
        \Illuminate\Support\Facades\Cache::clear();

        file_put_contents($this->tempDir.'/foo.yaml', '');

        $this->assertNotNull($one = $this->store->getItem('foo'));
        $this->store->forgetItem('foo');

        $this->assertNotNull($two = $this->store->getItem('foo'));
        $this->assertNotSame($one, $two);
    }

    /** @test */
    public function items_are_the_same_instance_every_time()
    {
        config(['cache.default' => 'file']); // Doesn't work when they're arrays since the object is stored in memory.
        \Illuminate\Support\Facades\Cache::clear();

        file_put_contents($this->tempDir.'/foo.yaml', '');

        $this->assertNotNull($one = $this->store->getItem('foo'));
        $this->assertNotNull($two = $this->store->getItem('foo'));
        $this->assertSame($one, $two);
    }
```

Reply:
Hmmm, I'm not sure we'd want to clear the cache between the getItem calls - as I don't think this would happen in the actual app?

Reply:
I'd imagine that in the actual app, you'd want the same instance to be returned, but I could be wrong.

https://github.com/statamic/cms/commit/a364f472e66c49118d9f08c0707b71de63a14190 Seems to suggest that would also be the desired outcome.

While https://github.com/statamic/cms/commit/20d2c4cc50d316368e888d0882e17f2b6c7dc6c2 specifically tests that they arent.

@jasonvarga are you able to elaborate on this?
Would you foresee any issues with clearing the store inbetween to force them to be different instances?

Reply:
I understand that with your blink changes, that it now returns the same instance every time.

I'm saying - we _need_ it to be a different instance every time, because of the reasoning explained in 20d2c4c. Your implementation will also need to be changed.

Think about Eloquent. Every time you do `Post::find(1)`, it gets a fresh version from the DB.

Maybe you can keep blink, but do a `clone` so it's a different instance.
i.e. `return clone Blink::once($cacheKey, ...`

Reply:
Ahhh okay, just clocked it, that make more sense now. Cloning seems like the best way to do it, though, I believe this might also require some changes to other tests.

Seen as though the `array` cache driver is used in the tests, getCachedItem() will return the same object, as it's in memory. But if we use blink with clone, then getCachedItem() will return a different instance (like we'd expect).

For this reason, I think some unit tests which compare items from the stache would need changing from assertSame() to assertEquals(). Would you agree?

<img width="867" alt="Screenshot 2021-07-15 at 16 06 24" src="https://user-images.githubusercontent.com/25618897/125811268-c0e81531-5080-402e-a95b-2b6844d742e3.png">

Reply:
Yeah I think thats okay there